demisto_sdk/commands/lint/linter.py

```diff
@@ -1306,3 +1347,290 @@ def _get_commands_list(self, script_obj: dict):
+            logger.error(err_msg)
+            raise Exception(err_msg)
+
+    def _get_versioned_native_image(
+        self,
+        native_image: str,
+    ) -> Union[str, None]:
+        """
+        Gets a versioned native image name, and finds it's reference (tag) in the docker_native_image_config.json file.
+        Args:
+            native_image (str): Name of the Native image to run on.
+        Returns (str): The native image reference (tag).
+        """
+        logger.info(
+            f"{self._pack_name} - Get Versioned Native Image - {native_image} - Started"
+        )
+
+        native_image_config = (
+            NativeImageConfig()
+        )  # parsed docker_native_image_config.json file (a singleton obj)
+
+        return native_image_config.get_native_image_reference(native_image)
+
+    def _get_all_docker_images(
+        self,
+        script_obj: Dict,
+        script_id: str,
+        supported_native_images: Set[str],
+    ) -> List[str]:
+        """
+        Gets the following docker images references:
+            1. Native GA - the native image of the current server version.
+            2. Native Maintenance - the native image of the previous server version.
+            3. Native Dev - The latest tag of the native image for Docker Hub.
+            4. The docker images that appear in the YML file of the integration/script that lint runs on.
+        Args:
+            script_obj (Dict): A yml as dict of the integration/script that lint runs on.
+            script_id (str): The ID of the integration/script that lint runs on.
+            supported_native_images (set): A set including the names of the native images supported for the
+                                           script/integration that lint runs on.
+        Returns (List): A list including all the docker images to run on.
+        """
+        logger.info(f"{self._pack_name} - Get All Docker Images Started")
+
+        # Get docker images from yml:
+        logger.info(f"{self._pack_name} - Get Docker Image from YML - Started")
+        imgs = get_docker_images_from_yml(script_obj)
+
+        # Get native images:
+        native_image_config = (
+            NativeImageConfig()
+        )  # parsed docker_native_image_config.json file (a singleton obj)
+
+        for native_image in native_image_config.native_images:
+            if self._is_native_image_support_script(
+                native_image, supported_native_images, script_id
+            ):
+
+                if native_image == DockerImageFlagOption.NATIVE_DEV.value:
+                    #  Get native latest from Docker Hub
+                    native_image_ref = self._get_dev_native_image(script_id)
+                else:  # versioned native image
+                    native_image_ref = self._get_versioned_native_image(native_image)
+
+                if native_image_ref:
+                    imgs.append(native_image_ref)
+
+        return imgs
+
+    def _get_docker_images_for_lint(
```

Comment:
this is a very long function, can it be split to smaller ones?

Reply:
@dorschw - I took out another part and created a separate function... but the code is already split into several auxiliary functions... can you have a second look, please?

Reply:
much better :)